aws/table_aws_sfn_state_machine_execution_history.go

```diff
@@ -287,8 +293,22 @@ func listStepFunctionsStateMachineExecutionHistories(ctx context.Context, d *plu
 	executionCh := make(chan []historyInfo, len(executions))
 	errorCh := make(chan error, len(executions))
 
-	// Iterating all the available executions
+	// Iterating all the available executions matching the query quals, if any
```

Comment:
If we're given an `execution_arn` key qual, I wonder if this function even needs to call the `ListExecutions` API method?

Since we'd have a specific `execution_arn` when we enter this function, we could instead just call `GetExecutionHistory` once, passing in the given `execution_arn` and then returning the history data (or 0 rows if there is no history data).

If we don't get an `execution_arn`, then we would rely on listing all executions and then getting the histories for each of them (like the function does today).

This isn't normally how we'd want to do it, but due to the lack of nested parent hydrates, this may be a way to reduce the number of API calls we do.

@pdecat Does this approach seem like something that would work and cut down on total number of requests when `execution_arn` is passed in?

Reply:
Good point! I've implemented what you suggested, PTAL.